File: src/org/jetbrains/haskell/cabal/highlight/CabalHighlighterFactory.java
package org.jetbrains.haskell.cabal.highlight;

import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.haskell.highlight.HaskellHighlighter;

public class CabalHighlighterFactory extends SingleLazyInstanceSyntaxHighlighterFactory {

  @NotNull
  protected CabalHighlighter createHighlighter() {
    return new CabalHighlighter();
  }

}

File: plugin/src/org/jetbrains/cabal/highlight/CabalHighlighterFactory.kt
package org.jetbrains.cabal.highlight

import com.intellij.openapi.fileTypes.SingleLazyInstanceSyntaxHighlighterFactory

class CabalHighlighterFactory : SingleLazyInstanceSyntaxHighlighterFactory() {

    override fun createHighlighter(): CabalHighlighter {
        return CabalHighlighter()
    }

}